clang/lib/Lex/Pragma.cpp

```diff
@@ -987,8 +987,10 @@ bool Preprocessor::LexOnOffSwitch(tok::OnOffSwitch &Result) {
 
   // Verify that this is followed by EOD.
   LexUnexpandedToken(Tok);
-  if (Tok.isNot(tok::eod))
+  if (Tok.isNot(tok::eod)) {
     Diag(Tok, diag::ext_pragma_syntax_eod);
+    DiscardUntilEndOfDirective();
```

Comment:
Found why I added that. Without the call to DiscardUntilEndOfDirective the LIT test Preprocessor/pragma_unknown.c will fail with unknown type name 'POWER' for this case: #pragma STDC CX_LIMITED_RANGE ON FULL POWER

Reply:
Ah! That seems to be an existing issue that's unrelated to your patch: https://godbolt.org/z/Mfz5W9M4P -- so I'd say this should probably be done as a separate change rather than rolled into this one. Then you can test all the uses of `on-off-switch` pragmas to make sure all of them behave properly in the presence of extra tokens.

Reply:
That error sounds like we're resuming parsing in the middle of the directive, which is definitely inappropriate.  But I'm surprised that this has to be handled here as opposed to being done systematically for all pragmas.  This is essentially just an "unexpected extra tokens after pragma" diagnostic, right?  Seems like it should be generic.

Reply:
Yes, it's actually generic. Any other pragma with additional unexpected token is giving the warning and an error. So I guess I will remove the code and have it error on the LIT test. This can be fixed in another PR. Thanks.